src/Routing/Types.purs

```diff
@@ -0,0 +1,6 @@
+module Routing.Types where
+
+import qualified Data.StrMap as M
+
+data RoutePart = Path String | Query (M.StrMap String)
+type Route = [RoutePart]
```

Comment:
This representation doesn't seem to represent the fact that you only have one `Query` part. Why not

```purescript
data Route = Route { parts :: List String, params :: M.StrMap String }
```